### DISCUSSION TITLE
Confirming that Observable Sets don't work with Object References

### DISCUSSION THREAD

**Discussion by focomoso** [mobxjs/mobx#3549] [Q&A] [answered]
From here: https://github.com/mobxjs/mobx/issues/69#issuecomment-456106054 it looks like observable sets are supported, but I can't find any documentation on them.

And from earlier in that same thread it's suggested that sets would have to be of strings (not object references): https://github.com/mobxjs/mobx/issues/69#issuecomment-360433207

Trying it out seems to confirm this:

```
import { makeAutoObservable } from 'mobx';

export default class Store {
  set = new Set();

  constructor() {
    makeAutoObservable(this);
    console.log(this);

    const obj = {a: 'a'};

    this.set.add(obj);
    console.log(this.set.has(obj)); // false

    const str = 'abc';
    this.set.add(str);
    console.log(this.set.has(str)); // true

    const n = 3;
    this.set.add(n);
    console.log(this.set.has(n)); // true
  }
}
```
Just want to confirm that this is intended behavior and suggest that it would be good to add to the docs somewhere if it is.

**Comment by urugator** [comment, marked as answer]
`ObservableSet` works with anything a `Set` does, but same as any observable collection it's `deep` by default, so it first converts the value to `observable`, which creates a copy - so it actually stores a different reference. I agree, it's confusing.
Either make the object observable before adding it to the set or make the set shallow.

**Comment by focomoso** [comment]
Ahh. This makes sense. The copying of non-observable objects slipped through my reading. Not sure whether it's because I read up too fast or if it's not clear in the docs. Thanks.
